### PR TITLE
[1LP][RFR] Added NavigatableMixin to collections

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================
Added NavigatableMixin to the Collection classes
{{ pytest: cfme/tests/cloud/test_tenant.py cfme/tests/cloud/test_stack.py cfme/tests/cloud/test_keypairs.py -vvvv }}

### CODE REVIEW

File: cfme/cloud/keypairs.py [hunk 1]
```diff
@@ -110,14 +110,13 @@ def is_displayed(self):
...
     """ Collection object for the :py:class: `cfme.cloud.KeyPair`. """
+    def __init__(self, appliance):
+        self.appliance = appliance
 
     def instantiate(self, name, provider, public_key=None, appliance=None):
-        return KeyPair(name, provider,
-                       public_key=public_key or "",
-                       appliance=appliance or self.appliance,
-                       collection=self)
+        return KeyPair(name, provider, self, public_key=public_key or "")
```
Comment (izapolsk) [inline cfme/cloud/keypairs.py]: REQUIRED: please put parameter names in calls. it makes code more readable and allows to avoid mistakes.

File: cfme/cloud/stack.py [hunk 2]
```diff
@@ -252,15 +252,14 @@ def is_displayed(self):
...
     """Collection class for cfme.cloud.stack.Stack"""
 
-    def __init__(self, appliance=None):
+    def __init__(self, appliance):
         self.appliance = appliance
-        Navigatable.__init__(self, appliance=self.appliance)
 
     def instantiate(self, name, provider, quad_name=None):
-        return Stack(name, provider, quad_name=quad_name, collection=self)
+        return Stack(name, provider, self, quad_name=quad_name)
```
Comment (izapolsk) [inline cfme/cloud/stack.py]: ^^

File: cfme/cloud/stack.py [hunk 3]
```diff
@@ -292,16 +291,16 @@ def delete(self, *stacks):
             raise ValueError('Some Stacks not found in the UI')
 
 
-class Stack(Pretty, Navigatable):
+class Stack(Pretty, NavigatableMixin):
     _param_name = "Stack"
     pretty_attrs = ['name']
 
-    def __init__(self, name, provider, quad_name=None, collection=None):
+    def __init__(self, name, provider, collection, quad_name=None):
```
Comment (izapolsk) [inline cfme/cloud/stack.py]: quad_icon should be removed but maybe in another PR.

File: cfme/tests/infrastructure/test_datastore_analysis.py [hunk 4]
```diff
@@ -29,15 +29,15 @@
 )
 
 
-def pytest_generate_tests(metafunc):
+def pytest_generate_tests(metafunc, appliance):
```
Comment (RonnyPfannschmidt) [inline cfme/tests/infrastructure/test_datastore_analysis.py]: this breaks the world

Comment (izapolsk) [reply cfme/tests/infrastructure/test_datastore_analysis.py]: oh, my bad.  but last PRT looked fine
